packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -1277,7 +1277,9 @@ class LocalizationsGenerator {
     // A pubspec.yaml file is required when using a synthetic package. If it does not
     // exist, create a blank one.
     if (useSyntheticPackage) {
-      final Directory syntheticPackageDirectory = _fs.directory(_defaultSyntheticPackagePath(_fs));
+      final Directory syntheticPackageDirectory = projectDirectory != null
+          ? projectDirectory!.childDirectory(_defaultSyntheticPackagePath(_fs))
+          : _fs.directory(_defaultSyntheticPackagePath(_fs));
```

Comment:
Without this change, this always created the `pubspec.yaml` using a relative path from the current directory (which when running the test, was the real `flutter_tools` folder, and not where the test project was being created). It felt like I should reuse `outputDirectory`, but that's set to a sub-folder of the package rather than the package root. Perhaps it's worth adding a `syntheticPackageDirectory` at the class level and moving this to the constructor alongside `outputDirectory`?